orangecontrib/text/tests/test_pubmed.py

```diff
@@ -0,0 +1,452 @@
+import json
+import os
+import unittest
+from unittest.mock import patch
+
+import numpy as np
+
+from orangecontrib.text.pubmed import (
+    Pubmed, PUBMED_TEXT_FIELDS,
+    _join_if_list, _mesh_headings_to_class,
+    _date_to_iso, _corpus_from_records,
+    _records_to_corpus_entries
+)
+
+
+class MockOSPath:
+    @staticmethod
+    def exists(path):
+        raise Exception()
+
+
+class MockEntrezHandle:
+    @staticmethod
+    def close():
+        return
+
+
+class MockEntrez:
+    """Used to mock Entrez/Medline reading/parsing methods.
+
+    Mocks read (after esearch and epost) and parse (after efetch).
+    """
+
+    def __init__(self, cache):
+        self.mock_responses = {}
+        with open(cache, 'r') as f:
+            self.mock_responses = json.loads(f.read())
+
+    def esearch(self, db, term, **keywds):
+        return MockEntrezHandle()
+
+    def read(self, handle):
+        return self.mock_responses.get('read')
+
+    def efetch(self, db, **keywords):
+        return MockEntrezHandle()
+
+    def epost(self, db, **keywds):
+        return MockEntrezHandle()
+
+    def parse(self, handle):
+        return self.mock_responses.get('parse')
+
+    # Exception mocking.
+    def esearch_exception(self, db, term, **keywds):
+        raise Exception
+
+    def efetch_exception(self, db, **keywds):
+        raise Exception
+
+    def epost_exception(self, db, **keywds):
+        raise Exception
+
+
+CACHE = os.path.join(os.path.dirname(__file__), 'pubmed-cache.txt')
+mock_entrez = MockEntrez(CACHE)
+mock_os_path = MockOSPath()
+
+
+def error_callback(exception):
+    return
+
+
+def progress_callback():
+    return
+
+
+class PubmedTests(unittest.TestCase):
+    EMAIL = 'mockemail@mockdomain.com'
+
+    def setUp(self):
+        self.pubmed = Pubmed(
+                self.EMAIL,
+                progress_callback=progress_callback,
+                error_callback=error_callback
+        )
+
+    @patch('os.path.exists', mock_os_path.exists)
+    def test_pubmed_object_creation(self):
+        self.assertRaises(
+                Exception,
+                Pubmed,
+                'faulty_email'
+        )
+
+        self.assertWarns(
+            RuntimeWarning,
+            Pubmed,
+            self.EMAIL,
+        )
+
+    def test_join_if_list(self):
+        not_list_input = 'test_value'
+        list_input = [
+            'test',
+            'value',
+            'here',
+        ]
+
+        self.assertEqual(_join_if_list(not_list_input), not_list_input)
+        self.assertEqual(_join_if_list(list_input), 'test value here')
+
+    def test_mesh_headings_to_class(self):
```

Comment:
Add at least the two test (or more if you find any other bugs) for errors mentioned above. That is, one that has `Multi Word Mesh/...` and one without separators like in `Heading`.